src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -1056,5 +1057,31 @@ private boolean getMoreAvailable(Page<Organisation> pageableOrganisations) {
         return !pageableOrganisations.isLast();
     }
 
+    @Override
+    public ResponseEntity<ContactInformationResponse> updateContactInformationForOrganisation(
+        ContactInformationCreationRequest contactInfo, String organisationIdentifier) {
+
+        Organisation organisation = organisationRepository.findByOrganisationIdentifier(organisationIdentifier);
+        List<ContactInformation> existingContactInformationList = organisation.getContactInformation();
+        //if multiple contact informations are found they will all be updated
+        if (contactInfo != null) {
+            existingContactInformationList.forEach(existingInfo -> {
+                existingInfo.setAddressLine1(RefDataUtil.removeEmptySpaces(contactInfo.getAddressLine1()));
+                existingInfo.setAddressLine2(RefDataUtil.removeEmptySpaces(contactInfo.getAddressLine2()));
+                existingInfo.setAddressLine3(RefDataUtil.removeEmptySpaces(contactInfo.getAddressLine3()));
+                existingInfo.setTownCity(RefDataUtil.removeEmptySpaces(contactInfo.getTownCity()));
+                existingInfo.setCounty(RefDataUtil.removeEmptySpaces(contactInfo.getCounty()));
+                existingInfo.setCountry(RefDataUtil.removeEmptySpaces(contactInfo.getCountry()));
+                existingInfo.setPostCode(RefDataUtil.removeEmptySpaces(contactInfo.getPostCode()));
+                existingInfo.setOrganisation(organisation);
+                existingInfo.setLastUpdated(LocalDateTime.now());
+                ContactInformation savedContactInformation = contactInformationRepository.save(existingInfo);
+                addDxAddressToContactInformation(contactInfo.getDxAddress(), savedContactInformation);
```

Comment:
Update DX address details only when needed and this has to be decided from the request. 
To incorporate both contact information and DX address updates add query params to the request like below

1. contactInformationUpdate = true/false (required)
2. dxAddressUpdate=true/false (required)

And update when param value is true, and add validation like if param value is true but data is missing then return bad request otherwise update. And also return error message when both params are false.

Reply:
added a boolean dxAddressUpdateRequired in ContactInformationCreationRequest to decide if dx address needs to be updated .
a flag for contactInformationUpdate will not be required